graphx/src/main/scala/org/apache/spark/graphx/impl/VertexPartitionBaseOps.scala

```diff
@@ -136,6 +136,31 @@ private[graphx] abstract class VertexPartitionBaseOps
     leftJoin(createUsingIndex(other))(f)
   }
 
+  def leftJoinWithFold[VD2: ClassTag, VD3: ClassTag, A]
+      (other: Self[VD2], acc: A)
```

Comment:
I'm pretty sure @ankurdave was looking at the `*WithFold` style addition and not changing the original method names to keep backwards compatibility as best as possible. I agree its a bit confusing, but I figure maintaining backwards compatibility as best as best is the better option.